components/brave_rewards/core/wallet_provider/connect_external_wallet.cc

```diff
@@ -8,22 +8,31 @@
 #include <utility>
 
 #include "base/containers/contains.h"
+#include "base/functional/bind.h"
+#include "base/location.h"
+#include "base/task/sequenced_task_runner.h"
 #include "brave/components/brave_rewards/core/common/random_util.h"
 #include "brave/components/brave_rewards/core/database/database.h"
+#include "brave/components/brave_rewards/core/endpoints/request_for.h"
 #include "brave/components/brave_rewards/core/ledger_impl.h"
 #include "brave/components/brave_rewards/core/logging/event_log_keys.h"
 #include "brave/components/brave_rewards/core/logging/event_log_util.h"
 #include "brave/components/brave_rewards/core/wallet/wallet_util.h"
 
 namespace brave_rewards::internal {
-
+using endpoints::GetWallet;
 using endpoints::PostConnect;
+using endpoints::RequestFor;
 using wallet::GetWalletIf;
 
 namespace wallet_provider {
 
 ConnectExternalWallet::ConnectExternalWallet(LedgerImpl& ledger)
-    : ledger_(ledger) {}
+    : ledger_(ledger) {
+  base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
```

Comment:
I'm wondering if it's safe to trigger flows from constructors (even with a posted task), since database and state migrations may not have had a chance to run yet.